Dockerfile.task_base

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM python:3.9-slim
```

Comment:
Aligns this with the rest of our images.

At some point, I'd like to bump this (across the board) to 3.10 and then 3.11, but that's a dedicated task.